src/Umbraco.Web/Extensions/UdiExtensions.cs

```diff
@@ -36,8 +36,30 @@ public static IPublishedContent ToPublishedContent(this Udi udi)
                         return umbracoHelper.TypedMember(memberAttempt.Result);
                     break;
             }
-            
+
             return null;
         }
+
+        /// <summary>
+        /// An extension method to easily acquire the integer Id for a given udi
+        /// </summary>
+        /// <param name="udi"></param>
+        /// <returns>An <see cref="int"/> identifier if the item is found, -1 otherwise</returns>
+        public static int ToIntId(this Udi udi)
```

Comment:
I can live with `Umbraco.GetIdForUdi(...)` - that's the more "Umbraco" way of doing things.
Sure we can add the `Url` too, that's nice. Eventually anything that requires an int argument now also will accept a Udi.